WPFHospitalEditor/HospitalMap.xaml.cs

```diff
@@ -10,61 +10,69 @@
+            CanvasService.addObjectToCanvas(findOutterMapObjects(allMapObjects), canvas);
         }
-        private void selectBuilding(object sender, MouseButtonEventArgs e)
-        {
-            MapObject chosenBuilding = checkWhichObjectIsClicked(e, AllMapObjects.allOuterMapObjects);
-            if (chosenBuilding != null)
-            {
-                goToClickedBuilding(chosenBuilding);
-            }
-        }
-        public MapObject checkWhichObjectIsClicked(MouseButtonEventArgs e, List<MapObject> allMapObjectsShowed)
+
+        private List<MapObject> findOutterMapObjects(List<MapObject> allMapObjects)
         {
-            for (int i = 0; i < allMapObjectsShowed.Count; i++)
+            foreach (MapObject mapObject in allMapObjects)
             {
-                if (checkIfPointIsInRectangle(e, allMapObjectsShowed[i]))
+                if (mapObject.Description.Equals(""))
                 {
-                    return allMapObjectsShowed[i];
+                    allOuterMapObjects.Add(mapObject);
                 }
             }
-            return null;
+            return allOuterMapObjects;
         }
-        private Boolean checkIfPointIsInRectangle(MouseButtonEventArgs e, MapObject mapObject)
-        {
-            return (e.GetPosition(canvas).X > mapObject.MapObjectMetrics.MapObjectCoordinates.X
-                    && e.GetPosition(canvas).X < mapObject.MapObjectMetrics.MapObjectCoordinates.X + mapObject.MapObjectMetrics.MapObjectDimensions.Width
-                    && e.GetPosition(canvas).Y > mapObject.MapObjectMetrics.MapObjectCoordinates.Y
-                    && e.GetPosition(canvas).Y < mapObject.MapObjectMetrics.MapObjectCoordinates.Y + mapObject.MapObjectMetrics.MapObjectDimensions.Height);
+        private void selectBuilding(object sender, MouseButtonEventArgs e)
+        {           
+            MapObject chosenBuilding = CanvasService.checkWhichObjectIsClicked(e, allOuterMapObjects, canvas);
+            if (chosenBuilding != null && chosenBuilding.MapObjectType == MapObjectType.Building)
+            {
+                goToClickedBuilding(chosenBuilding);
+            }
         }
+       
         private void goToClickedBuilding(MapObject mapObject)
         {
-                canvas.Children.Clear();
-                Building window1 = new Building(mapObject.Id);
-                this.Close();
-                window1.ShowDialog();
-        }
-        public void addObjectToCanvas(List<MapObject> objectsToShow, Canvas canvas)
-        {
-            for (int i = 0; i < objectsToShow.Count; i++)
+            MapObjectController mapObjectController = new MapObjectController(new MapObjectService(new MapObjectRepository(new FileRepository(AllConstants.MAPOBJECT_PATH))));
+            List<MapObject> buildingObjects = new List<MapObject>();
+
+            foreach (MapObject mapObjectIteration in mapObjectController.getAllMapObjects())
```

Comment:
If all object have different id you could use Dictionary to save all MapObjects, so you can find object without iterating trough it.